### PR TITLE
Migrate release jobs to GitHub Actions

### PR DESCRIPTION
**Pull request by cPu1**
### Description

This changelist migrates the release and release candidate jobs from CircleCI to GitHub Actions.

Closes https://github.com/weaveworks/eksctl/issues/4731 

Closes https://github.com/weaveworks/eksctl/issues/4707

TODO:

- [ ] Slack notification

### Checklist
- [ ] ~Added tests that cover your change (if possible)~ Not applicable
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: build/scripts/do-release.sh [hunk 1]
```diff
@@ -1,26 +1,19 @@
 #!/bin/sh -ex
 
-if [ -z "${CIRCLE_PROJECT_REPONAME}" ] ; then
-  echo "Missing repo name, please set CIRCLE_PROJECT_REPONAME"
-  exit 1
+if [ -z "${GITHUB_REF_NAME}" ] || [ "${GITHUB_REF_TYPE}" != "tag" ] ; then
+    echo "Expected a tag push event, skipping release workflow";
+    exit 1;
```
[the remark this answers is not in the document]
Comment (cPu1) [reply build/scripts/do-release.sh:5]: It's out of habit, I'll remove them :slightly_smiling_face: . Thanks.

File: build/scripts/do-release-candidate.sh [hunk 2]
```diff
@@ -1,20 +1,17 @@
...
+if [ -z "${GITHUB_REF_NAME}" ] || [ "${GITHUB_REF_TYPE}" != "tag" ] ; then
+  echo "Expected a tag push event, skipping release workflow"
+  exit 1
+fi
 
-  if [[ ! -f "${RELEASE_NOTES_FILE}" ]]; then
-    echo "Release notes ${RELEASE_NOTES_FILE} not found. Exiting..."
-    exit 1
-  fi
+RELEASE_NOTES_FILE="docs/release_notes/${GITHUB_REF_NAME/-rc.*}.md"
```
Comment (nikimanoledaki) [inline build/scripts/do-release-candidate.sh:8]: we have not been appending `-rc` to the release notes, is it necessary? it looks like line 10 will fail if the file name doesn't match this pattern..

File: build/scripts/do-release-candidate.sh [hunk 3]
```diff
@@ -1,20 +1,17 @@
...
 
-  goreleaser release --rm-dist --timeout 60m --skip-validate --config=./.goreleaser.yml --release-notes="${RELEASE_NOTES_FILE}"
+if [ ! -f "${RELEASE_NOTES_FILE}" ]; then
+    echo "Release notes file ${RELEASE_NOTES_FILE} does not exist. Exiting..."
+    exit 1
+fi
 
-  sleep 90 # GitHub API resolves the time to the nearest minute, so in order to control the sorting oder we need this
+export RELEASE_DESCRIPTION="${GITHUB_REF_NAME}"
```
Comment (nikimanoledaki) [inline build/scripts/do-release-candidate.sh:15]: where is `RELEASE_DESCRIPTION` used?

File: build/scripts/do-release.sh [hunk 4]
```diff
@@ -1,26 +1,19 @@
...
+    echo "Expected a tag push event, skipping release workflow"
+    exit 1
 fi
 
-if [ -z "${CIRCLE_PULL_REQUEST}" ] && [ -n "${CIRCLE_TAG}" ] && [ "${CIRCLE_PROJECT_USERNAME}" = "weaveworks" ] ; then
-  export RELEASE_DESCRIPTION="${CIRCLE_TAG} (permalink)"
-  RELEASE_NOTES_FILE="docs/release_notes/${CIRCLE_TAG}.md"
+tag="${GITHUB_REF_NAME}"
+
+export RELEASE_DESCRIPTION="${tag} (permalink)"
```
Comment (nikimanoledaki) [inline build/scripts/do-release.sh:10]: same as previous comment - not sure where `RELEASE_DESCRIPTION` is used?

**Review by nikimanoledaki — APPROVED** [review]
a couple of questions/comments, otherwise LGTM! :)

File: build/scripts/do-release-candidate.sh (hunk 3 above)
Comment (cPu1) [reply build/scripts/do-release-candidate.sh:15]: It's used in the `.goreleaser.yml` template.

File: build/scripts/do-release.sh (hunk 4 above)
Comment (cPu1) [reply build/scripts/do-release.sh:10]: It's used in the `.goreleaser.yml` template.

File: build/scripts/do-release-candidate.sh (hunk 2 above)
Comment (cPu1) [reply build/scripts/do-release-candidate.sh:8]: Yes, it's necessary. We don't append `-rc` to the release notes file but we do for RC tags. `GITHUB_REF_NAME` for RCs will have the `-rc.*` suffix and this line removes that suffix to get the release notes filename.

Comment (nikimanoledaki) [reply build/scripts/do-release-candidate.sh:8]: Thanks for the explanation! 👍

**Review by aclevername — APPROVED** [review]

**Pull request merged**